core/src/main/scala/org/bykn/bosatsu/Declaration.scala

```diff
@@ -395,6 +475,8 @@ object Declaration {
           Pattern.StructKind.Named(nm, Pattern.StructKind.Style.TupleLike), Nil))
       case Var(v: Bindable) => Some(Pattern.Var(v))
       case Literal(lit) => Some(Pattern.Literal(lit))
+      case StringDecl(NonEmptyList(Right((_, s)), Nil)) =>
+        Some(Pattern.Literal(Lit.Str(s)))
```

Comment:
we need to be more careful, if we generate strings with patterns inside, they could get parsed as a string via Pattern parsing, so we probably need Pattern support to pass the tests.